WordPress/Classes/System/WordPressAppDelegate.swift

```diff
@@ -167,15 +167,12 @@ class WordPressAppDelegate: UIResponder, UIApplicationDelegate {
             bgTask = .invalid
         }
 
-        bgTask = app.beginBackgroundTask(expirationHandler: {
-            // Synchronize the cleanup call on the main thread in case
-            // the task actually finishes at around the same time.
-            DispatchQueue.main.async { [weak self] in
-                if let task = self?.bgTask, task != .invalid {
-                    DDLogInfo("BackgroundTask: executing expirationHandler for bgTask = \(task.rawValue)")
-                    app.endBackgroundTask(task)
-                    self?.bgTask = .invalid
-                }
+        bgTask = app.beginBackgroundTask(expirationHandler: { [weak self] in
+            // WARNING: The task has to be terminated immediately on expiration
+            if let task = self?.bgTask, task != .invalid {
+                DDLogInfo("BackgroundTask: executing expirationHandler for bgTask = \(task.rawValue)")
+                app.endBackgroundTask(task)
+                self?.bgTask = .invalid
```

Comment:
I agree with the removal of `DispatchQueue.main.async`, but do you have an idea why it was added in the first place? Maybe it was added to fix some bug. But even if there was a bug, the ideal solution shouldn't be `DispatchQueue.main.async`.

Reply:
I tried to check the history, but this code has been moved around a bit.

Based on the comment ("synchronize on the main thread"), this async is unnecessary because the expiration handler is already called on the main thread:

> The system calls the handler synchronously on the main thread, blocking the app’s suspension momentarily.
> From https://developer.apple.com/documentation/uikit/uiapplication/1623031-beginbackgroundtaskwithexpiratio